File: src/test/java/semicontinuity/idea/code/analyzer/graph/DAGraphViewRendererTest.java
package semicontinuity.idea.code.analyzer.graph;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import semicontinuity.idea.code.analyzer.graph.viewModel.vanilla.VFactory;

class DAGraphViewRendererTest implements DAGraphImplTestData1, DAGraphImplTestData3, DAGraphImplTestData4 {

    @Test
    void render1() throws JsonProcessingException {
        render(exampleGraph1());
    }

    @Test
    void render3() throws JsonProcessingException {
        render(exampleGraph3());
    }

    @Test
    void render4() throws JsonProcessingException {
        render(exampleGraph4());
    }

    private void render(DAGraph<String> graph) throws JsonProcessingException {
        var renderer = new DAGraphViewRenderer<>(graph, new VFactory<>(), (String id) -> id);

        var render = renderer.render();
        System.out.println(new ObjectMapper().writeValueAsString(render));
    }
}

File: src/test/java/semicontinuity/idea/code/analyzer/graph/DAGraphViewRendererTest.kt
package semicontinuity.idea.code.analyzer.graph

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import org.junit.jupiter.api.Test
import semicontinuity.idea.code.analyzer.graph.viewModel.vanilla.VComponent
import semicontinuity.idea.code.analyzer.graph.viewModel.vanilla.VFactory
import semicontinuity.idea.code.analyzer.graph.viewModel.vanilla.VIndependentComponents
import semicontinuity.idea.code.analyzer.graph.viewModel.vanilla.VLayer
import semicontinuity.idea.code.analyzer.graph.viewModel.vanilla.VMember
import semicontinuity.idea.code.analyzer.graph.viewModel.vanilla.VSplit
import java.util.function.Function

internal class DAGraphViewRendererTest : DAGraphImplTestData1, DAGraphImplTestData3, DAGraphImplTestData4 {
    @Test
    @Throws(JsonProcessingException::class)
    fun render1() {
        render(exampleGraph1())
    }

    @Test
    @Throws(JsonProcessingException::class)
    fun render3() {
        render(exampleGraph3())
    }

    @Test
    @Throws(JsonProcessingException::class)
    fun render4() {
        render(exampleGraph4())
    }

    @Throws(JsonProcessingException::class)
    private fun render(graph: DAGraph<String>) {
        val renderer =
            DAGraphViewRenderer(
                graph,
                VFactory(),
                { id: String -> id },
                { s: String -> s },
                DAGraphViewRendererDelegate(
                    VFactory(),
                    { id: String -> id },
                    { s: String -> s },
                )::doRender
            )

        val render = renderer.render()
        println(ObjectMapper().writeValueAsString(render))
    }
}
